cmd/state-svc/internal/deprecation/deprecation.go

```diff
@@ -94,17 +97,9 @@ func (checker *Checker) check(versionNumber string) (*Info, error) {
 		return nil, errs.Wrap(err, "Invalid version number: %s", versionNumber)
 	}
 
-	var infos []Info
-	if checker.shouldFetch() {
-		infos, err = checker.fetchDeprecationInfo()
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		infos, err = checker.cachedDeprecationInfo()
-		if err != nil {
-			return nil, errs.Wrap(err, "cachedDeprecationInfo failed")
-		}
+	infos, err := checker.cachedDeprecationInfo()
```

Comment:
I would just use the cache lib: http://github.com/ActiveState/cli/blob/80228c0a9d9c0c76daabe6d756eb027b1428b910/cmd/state-svc/internal/resolver/resolver.go#L40-L40

Much of the logic here can then probably be dumped, as it should just be a simple matter of "if cache is empty; refresh".